### PR TITLE
Feature/get latest analysis

### PR DESCRIPTION
**Pull request by shaytidhar**

### CODE REVIEW

File: intezer_sdk/analysis.py [hunk 1]
```diff
@@ -13,35 +14,38 @@
...
-        self._report = None  # type: dict
-        self._api = api or get_global_api()  # type: IntezerApi
-
-    def send(self, wait=False):  # type: (bool) -> None
-        if self.analyses_id:
+                 file_path: str = None,
+                 file_hash: str = None,
+                 file_stream: typing.BinaryIO = None,
+                 dynamic_unpacking: bool = None,
+                 api: IntezerApi = None,
```
Comment (davidt99) [inline intezer_sdk/analysis.py]: can you move it to be the last parameter

File: intezer_sdk/api.py [hunk 2]
```diff
@@ -63,20 +67,40 @@ def analyze_by_hash(self,
...
+        if file_path:
+            file_context_manager = open(file_path, 'rb')
+            file_name = os.path.basename(file_path)
+        elif file_stream:
+            file_context_manager = dummy_context_manager(file_stream)
+            file_name = 'file'
+        else:
+            raise ValueError('File path and file stream are mandatory')
+
         data = self._param_initialize(dynamic_unpacking, static_unpacking)
```
Comment (davidt99) [inline intezer_sdk/api.py]: if this line raises an exception, the context manager is not closed

File: intezer_sdk/api.py [hunk 3]
```diff
@@ -63,20 +67,40 @@ def analyze_by_hash(self,
...
-                        static_unpacking=None):  # type: (str,bool,bool) -> str
+                        file_path: str = None,
+                        file_stream: typing.BinaryIO = None,
+                        dynamic_unpacking: bool = None,
+                        static_unpacking: bool = None) -> str:
+        if file_path:
+            file_context_manager = open(file_path, 'rb')
+            file_name = os.path.basename(file_path)
+        elif file_stream:
+            file_context_manager = dummy_context_manager(file_stream)
```
Comment (davidt99) [inline intezer_sdk/api.py]: i don't like it... have a function that performs an analysis on a stream and reuse it when file path is received

File: intezer_sdk/utilities.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import contextlib
```
Comment (davidt99) [inline intezer_sdk/utilities.py]: this will have to go

File: intezer_sdk/analysis.py (hunk 1 above)
Comment (shaytidhar) [reply intezer_sdk/analysis.py]: Done

File: intezer_sdk/api.py (hunk 3 above)
Comment (shaytidhar) [reply intezer_sdk/api.py]: Done

File: intezer_sdk/utilities.py (hunk 4 above)
Comment (shaytidhar) [reply intezer_sdk/utilities.py]: Done

**Pull request merged**